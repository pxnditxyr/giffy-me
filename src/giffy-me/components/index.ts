export * from './AddNewCategory';
export * from './GifGrid';
export * from './GifItem';
