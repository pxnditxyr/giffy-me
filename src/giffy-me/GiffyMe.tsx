import { useState } from 'react';
import { AddNewCategory, GifGrid  } from './components';

export const GiffyMe = () => {


    const [ categories, setCategories ] = useState<Array<string>>([]);

    const onAddNewCategory = ( newCategory : string ) => {
        setCategories([
            newCategory,
            ...categories
        ])
    };

    return (
        <>
            <h1> Giffy Me </h1>
            <AddNewCategory
                currentCategories={ categories }
                onAddNewCategory={ onAddNewCategory }
                />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key={ category }
                            category={ category }
                            />
                    ))
                }
            </ol>
        </>
    );
};
