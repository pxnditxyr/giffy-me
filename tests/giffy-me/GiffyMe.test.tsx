import { render } from '@testing-library/react';
import { GiffyMe } from '../../src/giffy-me/GiffyMe';


describe( 'tests in GiffyMe component', () => {
    test( 'should render GiffyMe component', () => {
        const { container } = render( <GiffyMe /> );
        expect( container ).toMatchSnapshot();
    });
});
