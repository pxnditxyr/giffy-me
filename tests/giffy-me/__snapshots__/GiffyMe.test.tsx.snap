// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GiffyMe component should render GiffyMe component 1`] = `
<div>
  <main
    class="w-full flex justify-center items-center flex-col"
  >
    <h1
      class="text-8xl font-sans font-bold text-white py-8 text-center"
    >
       Giffy Me 
    </h1>
    <form
      aria-label="form"
      class="flex flex-col items-center justify-center w-full py-3 px-8"
    >
      <div
        class="flex flex-col gap-8"
      >
        <label
          class="text-amber-400 font-semibold text-3xl"
        >
           What are you thinking? 
        </label>
        <div
          class="flex flex-col gap-5 md:flex-row"
        >
          <input
            class="bg-gray-800 text-white font-semibold text-3xl py-2 px-4 rounded-lg w-full focus:outline-none md:w-2/3"
            name="category"
            placeholder="Add a new category"
            type="text"
            value=""
          />
          <button
            class="bg-gray-800 text-white font-semibold text-3xl py-2 px-4 rounded-lg w-full md:w-1/3"
            type="submit"
          >
             Search 
          </button>
        </div>
      </div>
    </form>
    <ol
      class="w-full"
    />
  </main>
</div>
`;
